src/components/Checkout/CheckoutDropInPaymentWrapper.vue

```diff
@@ -339,39 +340,61 @@ export default {
 					},
 				})
 				.then(kivaBraintreeResponse => {
+					// extract transaction saga id or transaction id from response
+					const transactionResult = this.useAsyncCheckout
+						? kivaBraintreeResponse?.data?.shop?.doNoncePaymentDepositAndCheckoutAsync
+						: kivaBraintreeResponse?.data?.shop?.doNoncePaymentDepositAndCheckout;
+
+					if (this.useAsyncCheckout && typeof transactionResult !== 'object') {
+						pollForCheckoutStatus(this.apollo, transactionResult)
```

Comment:
What happens if the user navigates away while this is polling? Will the complete transaction logic never run?

Reply:
This process uses the same page level "loading" state as the existing checkout mutation. So it's certainly possible for someone to decide not to wait on either flow to complete. In either case the checkout can complete in the background even if the post-process or thanks page are somehow blocked from occurring.

Reply:
So nothing from the UI is needed to "complete" the transaction? It should complete fine in the background?

Reply:
Correct, we've already called one of the checkout mutations at this point and gotten it's response. We are now just checking for the status IF we've used the async version.

Reply:
But yea I do see some potential cross over between the processes so will think on that some more.